src/encoded/types/analyte.py

```diff
@@ -15,3 +35,64 @@ class Analyte(SubmittedItem):
     item_type = "analyte"
     schema = load_schema("encoded:schemas/analyte.json")
     embedded_list = []
+
+    class Collection(Item.Collection):
+        pass
+
+
+def validate_rna_molecule_properties(context,request):
+    """Check that `molecule` contains RNA if RNA-specific properties have values.
+    
+    RNA-specific properties in `RNA_PROPERTIES`.
+    """
+
+    data = request.json
+    if 'molecule' in data:
+        if 'RNA' not in data['molecule']:
+            for property in RNA_PROPERTIES:
+                if data.get(property,""):
```

Comment:
Very minor comment: I generally leave off the default ("") value, defaults to None (just made me think at first "" was somehow significant).